src/cacheService.js

```diff
@@ -67,6 +70,23 @@ class CacheService {
     if (!cacheSpace) this.cache.write(`${rootStoreAddress}_${spaceName}`, space)
   }
 
+  async getThread (req, res, next) {
+    const spaceName = req.query.space
+    const threadName = req.query.name
+
+    const fullName = namesTothreadName(spaceName, threadName)
+    const cachePosts = await this.cache.read(fullName)
+    let posts
+    try {
+      posts = cachePosts || await this.pinning.getThread(fullName)
+    } catch (e) {
+      res.status(500).send('Error: Failed to load posts')
```

Comment:
can we differentiate between no thread/empty and code failure for error codes?

Reply:
Hm, I'm not sure. If it's a thread that noone has posted to we will still open it and it will be empty.

Reply:
ah right, so this 500 is for code failure? thats good then